tests/test_claude_wrapper_mutual_shim_loop.py

```
@@ -941,12 +947,162 @@ def test_interactive_spawning_shim_chain_refreshes_claude_pid(failures: list[str
         if settings_index + 1 >= len(args):
             failures.append(f"expected --settings value after spawning shim chain, got: {combined_output!r}")
             return
-        settings = json.loads(args[settings_index + 1])
+        settings = load_settings_value(args[settings_index + 1])
         hooks = settings.get("hooks", {})
         if len(hooks.get("SessionStart", [])) != 1 or len(hooks.get("Stop", [])) != 3:
             failures.append(f"expected one hook injection after spawning shim chain, got: {settings!r}")
 
 
+def test_custom_path_reentry_converges_to_one_settings_block(failures: list[str]) -> None:
```

Ruff (0.16.1)

[warning] Too many statements (51 > 50)

(PLR0915)